aws/alarms/lambda/notify_slack/notify_slack.ts

```diff
@@ -0,0 +1,159 @@
+import { gunzip } from "zlib";
+import { sendToSlack, sendToOpsGenie } from "./utils.js";
+
+export const handler = async (event: any) => {
+  console.log(JSON.stringify(event));
+  try {
+    if (event.awslogs) {
+      await handleCloudWatchLogEvent(event);
+    } else if (event?.Records?.[0]?.Sns?.Message) {
+      handleSnsEventFromCloudWatchAlarm(event);
+    } else {
+      console.log("No supported event type found.");
+      sendToSlack("Unknown Event", JSON.stringify(event, null, 2), "info");
+    }
+
+    return {
+      statusCode: "SUCCESS",
+    };
+  } catch (error) {
+    console.log("Handler Error: ", error);
+    return {
+      statusCode: "ERROR",
+      error: (error as Error).message,
+    };
+  }
+};
+
+/**
+ * Some log contains metadata information like `2023-07-13T13:48:30.151Z  535741e0-25a9-4c8b-a0a1-25d2b24bf1b4  INFO`
+ * which we want to ignore.
+ *
+ * @returns JSON object or false if the message is not JSON
+ */
+export const safeParseLogIncludingJSON = (message: string) => {
+  try {
+    const jsonPartOfLogMessage = message.slice(message.indexOf("{"));
+    return JSON.parse(jsonPartOfLogMessage);
+  } catch (e) {
+    return false;
+  }
+};
+
+export const getAlarmDescription = (message: string): string => {
+  try {
+    const parsedMessage = JSON.parse(message);
+    return parsedMessage.AlarmDescription ? parsedMessage.AlarmDescription : parsedMessage;
+  } catch (err) {
+    return message;
+  }
+};
+
+/**
+ * @returns the severity level: [error, warning, info, alarm_reset] based on the message
+ */
+export const getSNSMessageSeverity = (message: string): string => {
+  const errorMessages = ["error", "critical"];
+  const warningMessages = ["warning", "failure"];
+  const alarm_ok_status = '"newstatevalue":"ok"'; // This is the string that is returned when the alarm is reset
+
+  message = message.toLowerCase();
+
+  if (message.indexOf("sev1") != -1) return "SEV1";
+
+  for (var errorMessagesItem in errorMessages) {
```

Comment:
Did not add a comment for all occurrences but `var` can be replaced by `const` in most cases now that we are using Typescript